### PR TITLE
[YAML] Modified ICMD scripts Aug 30

### PR DESCRIPTION
**Pull request by manjunath-grl**
`Fixes ` : [321](https://github.com/project-chip/matter-test-scripts/issues/321)

Modified ICMD scripts
```
TC-ICDM-4.1
TC-ICDM-5.1
TC-ICDM-5.2
TC-ICDM-6.1
```

### CODE REVIEW

File: src/app/tests/suites/certification/Test_TC_ICDM_5_2.yaml [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+          ./chip-tool icdmanagement read registered-clients 1 0
+          [1718188779.575015][1538:1540] CHIP:TOO: Endpoint: 0 Cluster: 0x0000_0046 Attribute 0x0000_0003 DataVersion: 2743878261
+          [1718188779.577470][1538:1540] CHIP:TOO:   RegisteredClients: 0 entries
+          [1718188779.577928][1538:1540] CHIP:EM: <<< [E:53088i S:19826 M:50965508 (Ack:261929127)] (S) Msg TX to 1:0000000000000001 [7914] [UDP:[fe80::e65f:1ff:fe49:ae1a%eth0]:5540] --- Type 0000:10 (SecureChannel:StandaloneAck)
+
+
+          If any entries are present, please send the command below to remove them:
+
+          ./chip-tool icdmanagement unregister-client 1<checkinnodeid> 1 0
+      disabled: true
```
Comment (andy31415) [inline src/app/tests/suites/certification/Test_TC_ICDM_5_2.yaml:43]: This seems to be adding new manual tests.

new manual test addition is not allowed anymore.

**Review by andy31415 — CHANGES_REQUESTED** [review]
We are not accepting additional manual tests in the SDK.

File: src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml [hunk 2]
```diff
@@ -24,122 +24,128 @@ config:
     endpoint: 0
 
 tests:
-    - label:
-          "Step 1: DUT issues an C_REGISTER_CLIENT command to the Test Harness."
-      PICS: ICDM.S.C00.Tx
+    - label: "Step 1a: TH reads from the DUT the RegisteredClients attribute."
```
Comment (yinyihu-silabs) [inline src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml:27]:
```suggestion
    - label: "Step 1: TH reads from the DUT the RegisteredClients attribute."
```

File: src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml [hunk 3]
```diff
@@ -24,122 +24,128 @@ config:
...
+          [1718186254.440634][2107:2109] CHIP:TOO: Endpoint: 0 Cluster: 0x0000_0046 Attribute 0x0000_0003 DataVersion: 547950372
+          [1718186254.440702][2107:2109] CHIP:TOO:   RegisteredClients: 0 entries
+          [1718186254.440975][2107:2109] CHIP:EM: <<< [E:34037i S:6387 M:176502552 (Ack:72711783)] (S) Msg TX to 1:0000000000000001 [2695] [UDP:[fe80::e65f:1ff:fe49:ae1a%eth0]:5540] --- Type 0000:10 (SecureChannel:StandaloneAck)
       disabled: true
 
-    - label:
-          "Step 2: DUT issues an C_UNREGISTER_CLIENT command to the Test
-          Harness."
-      PICS: ICDM.S.C02.Tx
+    - label: "Step 1b: TH reads from the DUT the OperatingMode attribute."
```
Comment (yinyihu-silabs) [inline src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml:57]:
```suggestion
    - label: "Step 2a: TH reads from the DUT the OperatingMode attribute."
```

File: src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml [hunk 4]
```diff
@@ -24,122 +24,128 @@ config:
...
+      verification: |
+          ./chip-tool icdmanagement read operating-mode 1 0
+          On TH, Verify that the DUT response contains value of OperatingMode=0 (SIT Operating Mode).
+          [1718186318.144714][2111:2113] CHIP:TOO: Endpoint: 0 Cluster: 0x0000_0046 Attribute 0x0000_0008 DataVersion: 547950372
+          [1718186318.144768][2111:2113] CHIP:TOO:   OperatingMode: 0
+          [1718186318.145033][2111:2113] CHIP:EM: <<< [E:841i S:49562 M:173701653 (Ack:236448256)] (S) Msg TX to 1:0000000000000001 [2695] [UDP:[fe80::e65f:1ff:fe49:ae1a%eth0]:5540] --- Type 0000:10 (SecureChannel:StandaloneAck)
+          [1718186318.145179][2111:2113] CHIP:EM: Flushed pending ack for MessageCounter:236448256 on exchange 841i
+      disabled: true
+
+    - label: "Step 1c: Verify that the ICD DNS-SD TXT key is present."
```
Comment (yinyihu-silabs) [inline src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml:67]:
```suggestion
    - label: "Step 2b: Verify that the ICD DNS-SD TXT key is present."
```

File: src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml [hunk 5]
```diff
@@ -24,122 +24,128 @@ config:
...
+             address = [fe80::e65f:1ff:fe49:ae1a]
+             port = [5540]
+             txt = ["ICD=0" "SAT=5000" "SAI=500" "SII=5500"]
       disabled: true
 
     - label:
-          "Step 3: DUT issues an C_STAY_ACTIVE_REQUEST command to the Test
-          Harness."
-      PICS: ICDM.S.C03.Tx
+          "Step 2: TH sends RegisterClient command. - CheckInNodeID:
```
Comment (yinyihu-silabs) [inline src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml:80]:
```suggestion
          "Step 3a: TH sends RegisterClient command. - CheckInNodeID:
```

File: src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml [hunk 6]
```diff
@@ -24,122 +24,128 @@ config:
...
+          [1720525842.157] [3915:3917] [DMG] Received Command Response Data, Endpoint=0 Cluster=0x0000_0046 Command=0x0000_0001
+          [1720525842.157] [3915:3917] [TOO] Endpoint: 0 Cluster: 0x0000_0046 Command 0x0000_0001
+          [1720525842.157] [3915:3917] [TOO]   RegisterClientResponse: {
+          [1720525842.157] [3915:3917] [TOO]     ICDCounter: 107030765
+          [1720525842.157] [3915:3917] [TOO]    }
+          [1720525842.162] [3915:3917] [DMG] ICR moving to [AwaitingDe]
+          [1720525842.162] [3915:3917] [EM] <<< [E:37646i S:31390 M:250907743 (Ack:224253965)] (S) Msg TX to 1:0000000000000001 [D503] [UDP:[fe80::e65f:1ff:fe49:ae1a%eth0]:5540] --- Type 0000:10 (SecureChannel:StandaloneAck)
+      disabled: true
+
+    - label: "Step 3a: TH reads from the DUT the OperatingMode attribute."
```
Comment (yinyihu-silabs) [inline src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml:96]:
```suggestion
    - label: "Step 3b: TH reads from the DUT the OperatingMode attribute."
```

File: src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml [hunk 7]
```diff
@@ -24,122 +24,128 @@ config:
...
+          ./chip-tool icdmanagement read operating-mode 1 0
+          On TH, Verify that the DUT response contains value of OperatingMode=1 LIT Operating Mode).
+
+          [1718186495.396762][2125:2127] CHIP:DMG: }
+          [1718186495.396950][2125:2127] CHIP:TOO: Endpoint: 0 Cluster: 0x0000_0046 Attribute 0x0000_0008 DataVersion: 547950373
+          [1718186495.397008][2125:2127] CHIP:TOO:   OperatingMode: 1
+          [1718186495.397302][2125:2127] CHIP:EM: <<< [E:57791i S:63297 M:51633949 (Ack:59403812)] (S) Msg TX to 1:0000000000000001 [2695] [UDP:[fe80::e65f:1ff:fe49:ae1a%eth0]:5540] --- Type 0000:10 (SecureChannel:StandaloneAck)
+      disabled: true
+
+    - label: "Step 3b: Verify that mDNS is advertising ICD key."
```
Comment (yinyihu-silabs) [inline src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml:107]:
```suggestion
    - label: "Step 3c: Verify that mDNS is advertising ICD key."
```

File: src/app/tests/suites/certification/Test_TC_ICDM_5_2.yaml [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+    - ICDM.S.F02
+
+config:
+    nodeId: 0x12344321
+    cluster: "Basic Information"
+    endpoint: 0
+
+tests:
+    - label:
+          "Step 1a: TH1 reads from the DUT the RegisteredClients attribute.
```
Comment (yinyihu-silabs) [inline src/app/tests/suites/certification/Test_TC_ICDM_5_2.yaml:29]: Missing preconditions.

**Review by yinyi-hu — CHANGES_REQUESTED** [review]

File: src/app/tests/suites/certification/Test_TC_ICDM_5_2.yaml (hunk 1 above)
Comment (cjandhyala) [reply src/app/tests/suites/certification/Test_TC_ICDM_5_2.yaml:43]: some of these are old tests (<=1.3 release) that were manual, until those are automated we have to keep/update the manual tests.

File: src/app/tests/suites/certification/Test_TC_ICDM_5_2.yaml (hunk 8 above)
Comment (manjunath-grl) [reply src/app/tests/suites/certification/Test_TC_ICDM_5_2.yaml:42]: Updated

File: src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml (hunk 2 above)
Comment (manjunath-grl) [reply src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml]: Updated

File: src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml (hunk 3 above)
Comment (manjunath-grl) [reply src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml]: Updated

File: src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml (hunk 5 above)
Comment (manjunath-grl) [reply src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml]: Updated

File: src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml (hunk 6 above)
Comment (manjunath-grl) [reply src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml]: Updated

File: src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml (hunk 7 above)
Comment (manjunath-grl) [reply src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml]: Updated

File: src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml (hunk 4 above)
Comment (manjunath-grl) [reply src/app/tests/suites/certification/Test_TC_ICDM_5_1.yaml]: Updated

**Comment by yunhanw-google** [comment]
automated tests have coverred 4.1 and 5.1
TC-ICDM-4.1
TC-ICDM-5.1

File: src/app/tests/suites/certification/Test_TC_ICDM_5_2.yaml (hunk 1 above)
Comment (andy31415) [reply src/app/tests/suites/certification/Test_TC_ICDM_5_2.yaml:56]: This is a new file containing manual tests. This will not be accepted in the SDK.

**Pull request closed without merging**